lib/effects/scale_effect.dart

```diff
@@ -4,43 +4,52 @@ import 'package:meta/meta.dart';
 import '../extensions/vector2.dart';
 import 'effects.dart';
 
-class ScaleEffect extends PositionComponentEffect {
+class ScaleEffect extends SimplePositionComponentEffect {
   Vector2 size;
-  double speed;
-  Curve curve;
   Vector2 _startSize;
   Vector2 _delta;
 
   ScaleEffect({
     @required this.size,
-    @required this.speed,
-    this.curve,
+    double duration, // How long it should take for completion
+    double speed, // The speed of the scaling in pixels per second
```

Comment:
I would add to these comments the fact that it's one xor the other (not only to the docs)